src/components/DrinkDetails/DrinkDetails.js

```diff
@@ -1,97 +1,95 @@
-import React, { Component } from "react";
-import axios from "../../axiosCocktail";
-import classes from "./DrinkDetails.module.scss";
-import Aux from "../../hoc/Auxillary/Auxillary";
+import React, { Component } from 'react';
+import axios from '../../axiosCocktail';
+import classes from './DrinkDetails.module.scss';
+import Aux from '../../hoc/Auxillary/Auxillary';
 
 class DrinkDetails extends Component {
-    state = {
-        ingredients: [],
-        strInstructions: "",
-        strGlass: "",
-        strDrink: "",
-        idDrink: "",
-        strDrinkThumb: "",
-    };
+  state = {
+    ingredients: [],
+    strInstructions: '',
+    strGlass: '',
+    strDrink: '',
+    idDrink: '',
+    strDrinkThumb: '',
+  };
 
-    componentDidMount() {
-        console.log("[componentDidMount]", this.props.match.params.id);
-        if (!this.state.idDrink.length > 0) {
-            this.setState({ idDrink: this.props.match.params.id });
-            this.fetchDrinkData();
-        }
+  componentDidMount() {
+    console.log('[componentDidMount]', this.props.match.params.id);
+    if (!this.state.idDrink.length > 0) {
+      this.setState({ idDrink: this.props.match.params.id });
+      this.fetchDrinkData();
     }
+  }
 
-    fetchDrinkData = async () => {
-        let data = await axios.get(
-            "/v1/1/lookup.php?i=" + this.props.match.params.id
-        );
-        console.log("[fetchDrinkData]", data.data.drinks[0]);
-        data = data.data.drinks[0];
-
-        let ingredients = [];
-        for (let i = 1; i <= 15; i++) {
-            if (
-                data[`strIngredient${i}`] !== null &&
-                data[`strIngredient${i}`] !== ""
-            ) {
-                ingredients.push({
-                    measure: data[`strMeasure${i}`],
-                    ingredient: data[`strIngredient${i}`],
-                });
-            }
-        }
+  fetchDrinkData = async () => {
+    let data = await axios.get(
+      '/v1/1/lookup.php?i=' + this.props.match.params.id,
+    );
+    console.log('[fetchDrinkData]', data.data.drinks[0]);
+    data = data.data.drinks[0];
 
-        this.setState({
-            ingredients: ingredients,
-            strInstructions: data.strInstructions,
-            strDrink: data.strDrink,
-            strGlass: data.strGlass,
-            strDrinkThumb: data.strDrinkThumb,
+    let ingredients = [];
+    for (let i = 1; i <= 15; i++) {
```

Comment:
Hard coding the 15 is fine for now but what if you had more in the future? I may be wrong but I think you are looping through the `data` if that's the case then you could do something like
for (let i = 0; i <= data.length; i++) {
     ...................
}

Reply:
I don't know if there's a better way but the data as doesn't have an array of ingredients but strIngredient1, strIngredient2 etc... 

![Data](https://user-images.githubusercontent.com/27687165/94611018-275d8d80-026f-11eb-8457-3f3caaf39bf0.JPG)

Reply:
fair enough then maybe you cant on this one